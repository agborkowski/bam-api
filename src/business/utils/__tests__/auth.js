jest.mock('knex');
// import knex from 'knex';
import getViewerAndRoles, { assertIsBamer, getRolesByEmail } from '../auth';

describe('auth utils (unit)', () => {
  it('assertIsBammer should throw an error', () => {
    expect(() => assertIsBamer('george@bam.tech', [])).toThrowErrorMatchingSnapshot();
    expect(() => assertIsBamer('george@bam.tech', ['ANONYMOUS'])).toThrowErrorMatchingSnapshot();
    expect(() =>
      assertIsBamer('george@bam.tech', ['NOT_REGISTRED'])
    ).toThrowErrorMatchingSnapshot();
  });
  it('return bammer roles for bammer email', () => {
    expect(getRolesByEmail('george@bam.tech')).toEqual(['BAMER']);
    expect(getRolesByEmail('george@theodo.fr')).toEqual([]);
  });
});

describe.withTestDatabase('auth utils (integration)', () => {
  it('does return the right user / roles', async () => {
    const email = 'george2@bam.tech';
    const viewerAndRoles = await getViewerAndRoles({ email });
    expect(viewerAndRoles.roles).toEqual(['BAMER']);
    expect(viewerAndRoles.user.email).toEqual('george2@bam.tech');
  });
});
