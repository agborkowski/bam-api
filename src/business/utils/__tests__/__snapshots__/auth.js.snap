// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth utils (unit) assertIsBammer should throw an error 1`] = `"User should be a bamer"`;

exports[`auth utils (unit) assertIsBammer should throw an error 2`] = `"User should be a bamer"`;

exports[`auth utils (unit) assertIsBammer should throw an error 3`] = `"Please connect to use this functionality"`;
